### PR TITLE
=cls #17846 Use provided scope for the distributed-data dependency

### PR DESCRIPTION
**Pull request by patriknw**
The scope in the pom.xml was "compile".

### CODE REVIEW

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -136,7 +136,7 @@ object AkkaBuild extends Build {
     id = "akka-cluster-sharding",
     base = file("akka-cluster-sharding"),
     dependencies = Seq(cluster % "compile->compile;test->test;multi-jvm->multi-jvm",
-        persistence % "compile;test->provided", distributedData % "compile;test->provided", clusterTools)
+        persistence % "compile;test->provided", distributedData % "provided;test", clusterTools)
```
Comment (patriknw) [inline project/AkkaBuild.scala]: Now it actually becomes scope "test" in the pom.xml. It should be "provided" in the pom.xml, but I don't know how to do that. If I only use "provided" here it works, but then we can't run tests. Scope "test" is alright in the pom, so if we don't have any better suggestions we'll use that.

**Comment by patriknw** [comment]
Refs #17846

File: project/AkkaBuild.scala (hunk 1 above)
Comment (ktoso) [inline project/AkkaBuild.scala]: have you tried `provided;test->provided`?

Comment (ktoso) [inline project/AkkaBuild.scala]: (Can you add a comment here describing what happens in the pom? Maybe some sbt wizard could help here at some point)

**Comment by ktoso** [comment]
LGTM in any case (would like the TODO comment though, to remind us why/how it works)

File: project/AkkaBuild.scala (hunk 1 above)
Comment (patriknw) [inline project/AkkaBuild.scala]: yes, I think I tried that and several other combinations. I'll add the TODO

**Pull request merged**